pp_netlib/functions.py

```diff
@@ -210,3 +286,420 @@ def save_graph(graph, backend, outdir, file_name, file_format):
+
+            gt.graph_draw(mst, pos=pos, vertex_fill_color=mst.vertex_properties['plot_color'],
+                    output=graph2_file_name, output_size=(3000, 3000))
+
+def draw_nx_mst(mst, out_prefix, isolate_clustering, overwrite):
+    import networkx as nx
+    
+    import matplotlib.pyplot as plt
+
+    graph1_file_name = out_prefix + "_mst_stress_plot.png"
+    graph2_file_name = out_prefix + "_mst_cluster_plot.png"
+
+    if overwrite or not os.path.isfile(graph1_file_name) or not os.path.isfile(graph2_file_name):
+        sys.stderr.write("Drawing MST\n")
+        pos = nx.spring_layout(mst, seed=13)
+        if overwrite or not os.path.isfile(graph1_file_name):
+            deg = list(degree for (node, degree) in mst.degree(mst.nodes()))
+            deg = 4 * (np.sqrt(deg) * 0.5 + 0.4)
+            ebet = np.array(list((nx.edge_betweenness_centrality(mst)).values()))
+            ebet /= ebet.max() / 10.
+            plt.figure(figsize=(10,7))
+            nx.draw_networkx_nodes(mst, pos=pos, node_size=10*deg, node_color=deg)
+            nx.draw_networkx_edges(mst, pos=pos, edge_color=ebet, width=ebet)
+            plt.axis("off")
+            # plt.show()
+            plt.savefig(graph1_file_name)
+            plt.close()
+
+        if overwrite or not os.path.isfile(graph2_file_name):
+            cluster_fill = {}
+            for cluster in set(isolate_clustering.values()):
+                cluster_fill[cluster] = list(np.random.rand(3)) + [0.9]
+            for v in mst.nodes():
+                mst.nodes[v]["plot_color"] = cluster_fill[isolate_clustering[v]]
+
+            plt.figure(figsize=(10,7))
+            nx.draw_networkx_nodes(mst, pos=pos, node_size=100, node_color=[color for (node, color) in mst.nodes(data="plot_color")])
+            nx.draw_networkx_edges(mst, pos=pos)
+            plt.axis("off")
+            # plt.show()
+            plt.savefig(graph2_file_name)
+            plt.close()
+
+def gt_save_graph_components(graph, out_prefix, outdir):
+    import graph_tool.all as gt
+
+    component_assignments, component_hist = gt.label_components(graph)
+    for component_idx in range(len(component_hist)):
+        remove_list = []
+        for vidx, v_component in enumerate(component_assignments.a):
+            if v_component != component_idx:
+                remove_list.append(vidx)
+        G_copy = graph.copy()
+        G_copy.remove_vertex(remove_list)
+
+        save_graph(G_copy, "GT", outdir, out_prefix+"_component_"+str(component_idx + 1), ".graphml")
+        del G_copy
+
+def nx_save_graph_components(graph, out_prefix, outdir):
+    import networkx as nx
+
+    for idx, c in enumerate(nx.connected_components(graph)):
+        subgraph = graph.subgraph(c)
+        save_graph(subgraph, "NX", outdir, out_prefix+"_component_"+str(idx + 1), ".graphml")
+        del subgraph
+
+########################
+####   .METADATA    ####
+########################
+#TODO
+def gt_get_graph_data(graph):
```

Comment:
What is this and the next function used for?

Reply:
These functions get node and edge data from the graph; they are not used in this pr

Reply:
Are they used elsewhere?

Reply:
these functions are only used in the write_metadata method in the visualisation, so I'll remove them from this pr as well